mesh/mesh.go

```diff
@@ -790,11 +819,27 @@ func (msh *Mesh) accumulateRewards(l *types.Layer, params Config) {
 		log.Uint64("total_reward_remainder", blockTotalRewardMod.Uint64()),
 		log.Uint64("layer_reward_remainder", blockLayerRewardMod.Uint64()),
 	)
-	err := msh.writeTransactionRewards(l.Index(), ids, blockTotalReward, blockLayerReward)
+	//report the rewards for each coinbase and each smesherID within each coinbase
+	//this can be thought off as a partition of the reward amongst all the smesherIDs
+	//that added the Coinbase into the block
+	for account, smesherAccountEntry := range coinbasesAndSmeshers {
```

Comment:
I think this logic is wrong. The first map is a map from coinbase -> smesher -> number of blocks. This should be all the data you need to aggregate and report rewards. Why do you need to reference `smeshers` at all here? Can't you just loop through `coinbasesAndSmeshers` and report one reward for each (coinbase, smesher) tuple?

Reply:
The code is functionally correct because of the invariant, but I agree that its pretty unclean. I think with the new map, we can do away with the `smeshers` map altogether, since we only really need that when obtaining the array of `SmesherID`s for a given `Coinbase`. I think I'll remove it from the code

Reply:
Made the code cleaner